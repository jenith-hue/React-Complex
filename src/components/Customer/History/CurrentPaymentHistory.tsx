import {
  makeStyles,
  RACButton,
  RACCOLOR,
  RACTable,
  RACTableCell,
  RACTableRow,
  Typography,
} from '@rentacenter/racstrap';
import React, { useEffect, useState } from 'react';
import InfiniteScroll from 'react-infinite-scroll-component';
import { useLocation } from 'react-router-dom';
import {
  ApiStateWrapper,
  Loader,
} from '../../../common/ApiStateWrapper/ApiStateWrapper';
import { days, RECEIPT_TYPE } from '../../../constants/constants';
import {
  usePaymentHistory,
  usePaymentHistoryActions,
} from '../../../context/PaymentHistory/PaymentHistoryProvider';
import { CustomerLocationState } from '../../../types/types';
import {
  formatDateString,
  formatStringDateHoursAndMinutes,
} from '../../../utils/utils';
import { Document } from '../../Document/Document';

const useClasses = makeStyles((theme: any) => ({
  tableContentColor: {
    color: RACCOLOR.INDEPENDENCE,
  },
  tableRoot: {
    marginTop: theme.typography.pxToRem(16),
  },
  tableBody: {
    position: 'relative',
  },
  tableHead: {
    position: 'sticky',
    top: 0,
    zIndex: 2,
    background: 'white',
  },
  contentHeight: {
    height: theme.typography.pxToRem(400),
  },
}));

export const CurrentPaymentHistory = () => {
  const [receiptId, setReceiptId] = useState('');

  const {
    payments,
    total,
    numberOfLoadedPayments,
    hasApiError,
    loading,
    init,
  } = usePaymentHistory();

  const renderTableHead = () => (
    <>
      <RACTableCell>
        <Typography variant="h5">Date paid</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Day</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Time</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Receipt #</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Total Payment</Typography>
      </RACTableCell>
      <RACTableCell align="center">
        <Typography variant="h5">Next Due Date</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Days Ext</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5"># Days late</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Store</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Payment Origin</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Tender Type</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Net Rent</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Club</Typography>
      </RACTableCell>
    </>
  );

  const renderTableContent = () => (
    <>
      {payments?.map((dataObj, index) => {
        const paymentDate = new Date(dataObj.paymentDate || '');
        return (
          <RACTableRow key={index} hover backgroundColor="white">
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {formatDateString(dataObj.paymentDate)}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {days[paymentDate.getDay()]}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {formatStringDateHoursAndMinutes(dataObj.paymentDate)}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <RACButton
                variant="text"
                size="small"
                color="primary"
                onClick={() => {
                  // eslint-disable-next-line no-console
                  console.log('Implement mee');
                }}
              >
                <Typography
                  variant="body2"
                  color="primary"
                  onClick={() => {
                    setReceiptId(dataObj.receiptId as string);
                  }}
                >
                  {dataObj.receiptId}
                </Typography>
              </RACButton>
            </RACTableCell>
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {dataObj.paymentAmount !== 'null'
                  ? `$${dataObj.paymentAmount}`
                  : '-'}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <Typography
                align="center"
                variant="body2"
                className={classes.tableContentColor}
              >
                {formatDateString(dataObj.nextDueDate)}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {dataObj.daysExtension}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {dataObj.daysLate}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {dataObj.storeNumber}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {dataObj?.paymentOrigin?.description || ''}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {dataObj.paymentMethodType?.join(', ')}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {dataObj.netRentalRevenue !== 'null'
                  ? `$${dataObj.netRentalRevenue}`
                  : '-'}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {dataObj.clubPayment !== 'null'
                  ? `$${dataObj.clubPayment}`
                  : '-'}
              </Typography>
            </RACTableCell>
          </RACTableRow>
        );
      })}
    </>
  );
  const classes = useClasses();
  const { fetchPaymentHistory, resetPaymentData } = usePaymentHistoryActions();

  const initialLoad = () => {
    fetchPaymentHistory(true);
    return resetPaymentData;
  };

  // eslint-disable-next-line react-hooks/exhaustive-deps
  useEffect(initialLoad, []);

  const location = useLocation<CustomerLocationState>();
  useEffect(() => {
    if (!location?.state?.customer) return;
    initialLoad();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [location?.state?.customer]);

  return (
    <>
      {receiptId && (
        <Document
          title="Receipt"
          documentId={receiptId}
          typeId={RECEIPT_TYPE}
          onClose={() => setReceiptId('')}
        />
      )}
      <InfiniteScroll
        next={fetchPaymentHistory}
        dataLength={numberOfLoadedPayments}
        hasMore={numberOfLoadedPayments < total}
        height={330}
        loader={<Loader />}
      >
        <ApiStateWrapper
          loading={loading && init}
          hasApiError={hasApiError}
          response={payments}
          successContent={
            <RACTable
              bodyClasses={{ root: classes.tableBody }}
              headClasses={{ root: classes.tableHead }}
              classes={{ root: classes.tableRoot }}
              renderTableHead={renderTableHead}
              renderTableContent={renderTableContent}
            />
          }
          classes={{
            loader: classes.contentHeight,
            apiError: classes.contentHeight,
            noItems: classes.contentHeight,
          }}
        />
      </InfiniteScroll>
    </>
  );
};
