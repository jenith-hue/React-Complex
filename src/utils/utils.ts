import { AxiosResponse } from 'axios';
import { add, sub, format } from 'date-fns';

import { isEmpty } from 'lodash';
import {
  ALL_OPTION,
  CONTENT_TYPE,
  DATE_FORMAT,
  Language,
  STORE_NUMBER_KEY,
} from '../constants/constants';
import { ReferenceItem, RouteResponse } from '../types/types';
import * as crypto from 'crypto';

export const sanitizeURL = (url: string): string => {
  if (!url) {
    return '';
  }

  const badURLRegex = RegExp('^((https)|(http)):/{3,}');
  const isBadURL = badURLRegex.test(url);

  if (isBadURL) {
    return 'https://' + url.replace(badURLRegex, '');
  }

  return url;
};

export const abbrev = (strToCutOff: string): string => {
  if (strToCutOff && strToCutOff.length > 13) {
    return strToCutOff.substr(0, 13) + '...';
  }

  return strToCutOff;
};

export const getErrorMessage = (response?: AxiosResponse<any>) => {
  if (response?.status === 400 && response?.data.errors[0].code) {
    return response?.data.errors[0].message;
  }

  return null;
};

export const orderByField = (fieldName: string, arrayToOrder: any[]) => {
  if (!arrayToOrder || !arrayToOrder.length || !fieldName) return arrayToOrder;
  return arrayToOrder.sort((a, b) => (a[fieldName] > b[fieldName] ? 1 : -1));
};

export const orderByFieldAndValue = (
  value: string,
  fieldName: string,
  arrayToOrder: any[]
) => {
  if (!arrayToOrder || !arrayToOrder.length || !fieldName) return arrayToOrder;
  return arrayToOrder.sort((a) => (a[fieldName] === value ? -1 : 1));
};

export const mapReferenceResponse = (references: ReferenceItem[]) => {
  if (!references || !references.length) return references;
  return references.map((reference) => ({
    label: reference.description,
    value: reference.referenceCode,
    ...reference,
  }));
};

// Not deleting method, because might be useful for Route selection in customer info header.
export const mapRoutesResponse = (routes: RouteResponse[]) => {
  if (!routes || !routes.length) return routes;
  return routes.map((routeObj) => ({
    label: routeObj.routeDescription,
    value: routeObj.routeCode,
    ...routeObj,
  }));
};

export const addAllOption = (arrayWithOptions: any[]) => {
  if (!arrayWithOptions) {
    return [
      {
        label: ALL_OPTION,
        value: ALL_OPTION,
      },
    ];
  }

  return [
    {
      label: ALL_OPTION,
      value: ALL_OPTION,
    },
    ...arrayWithOptions,
  ];
};

export const pipe =
  (...fns: any) =>
  (x: any) =>
    fns.reduce((v: any, f: (value: any) => void) => f(v), x);

export const formatDate = (date: Date, dateFormat = DATE_FORMAT): string => {
  if (isValidDate(date)) return format(date, dateFormat);
  return '';
};

// Dates are stored with local timezone of the respective store.
// We want to show the same data, by ignoring the user's local date settings
// e.g. 2021-07-19T16:07:42 or 2021-07-20
// yyyy-mm-dd
export const formatDateString = (
  dateAsString: string | undefined,
  returnedFormat?: string
): string => {
  if (!dateAsString || typeof dateAsString !== 'string') {
    return '';
  }
  let dateWithoutHours;
  if (dateAsString.includes('T')) {
    dateWithoutHours = dateAsString.split('T')[0];
  } else {
    dateWithoutHours = dateAsString;
  }

  // [0] = 'yyyy'; [1] = 'mm'; [2] = 'dd'
  let dateAsArray = [] as string[];
  if (dateWithoutHours.includes('-')) {
    dateAsArray = dateWithoutHours.split('-');
  } else if (dateWithoutHours.includes('/')) {
    dateAsArray = dateWithoutHours.split('/');
  }

  if (dateAsArray?.length !== 3) {
    return '';
  }

  if (returnedFormat === Language.Spanish) {
    return `${dateAsArray[2]} de ${spanishMonths[dateAsArray[1]]} de ${
      dateAsArray[0]
    }`;
  }

  if (returnedFormat === 'MM-dd-yyyy') {
    return `${dateAsArray[1]}-${dateAsArray[2]}-${dateAsArray[0]}`;
  }
  // mm/dd/yyyy
  return `${dateAsArray[1]}/${dateAsArray[2]}/${dateAsArray[0]}`;
};

export const toLocaleStringWithMinutes = (date: Date): string => {
  if (isValidDate(date))
    return date.toLocaleString('en-US', {
      hour: 'numeric',
      minute: 'numeric',
      hour12: true,
    });

  return '';
};

// e.g. "2022-03-31T02:56:44.738-05:00"
export const formatStringDateHoursAndMinutes = (
  dateAsString: string | undefined
): string => {
  try {
    if (dateAsString && isValidDate(new Date(dateAsString))) {
      const split1 = dateAsString.split('T');
      if (!split1[1]) {
        return '';
      }
      const split2 = split1[1].split(':');
      if (!split2[1]) {
        return '';
      }

      const hours = Number(split2[0]);
      const minutes = split2[1];

      let hoursToDisplay = hours % 12;
      let meridian = 'AM';

      if (hours >= 12) {
        meridian = 'PM';
      }
      if (hoursToDisplay === 0) {
        hoursToDisplay = 12;
      }

      return `${hoursToDisplay}:${minutes} ${meridian}`;
    }

    return '';
  } catch {
    return '';
  }
};

export type ListStatus =
  | 'initial'
  | 'loading'
  | 'apiError'
  | 'empty'
  | 'success';
export interface ComponentStateProps {
  loading?: boolean;
  hasApiError?: boolean;
  response?: any;
}

export const getComponentState = ({
  loading,
  hasApiError,
  response,
}: ComponentStateProps): ListStatus => {
  if (loading) return 'loading';
  if (hasApiError) return 'apiError';

  if (response) {
    if (Array.isArray(response)) {
      return response.length > 0 ? 'success' : 'empty';
    } else if (isEmpty(response)) {
      return 'empty';
    }
    return 'success';
  } else {
    return 'initial';
  }
};

export const DAYS_OF_WEEK = [
  'Sunday',
  'Monday',
  'Tuesday',
  'Wednesday',
  'Thursday',
  'Friday',
  'Saturday',
];

export function isValidDate(d: any) {
  return d instanceof Date && !isNaN(d.getTime());
}

export const moveItemToTheEnd = (
  items: any,
  fieldName: string,
  value: string | number
): any => {
  if (!fieldName) return items;
  if (!value) return items;
  if (!Array.isArray(items)) return items;
  if (items.length < 2) return items;
  const itemsCopy = [...items];

  const index = itemsCopy.findIndex((item) => item[fieldName] === value);
  if (index < 0) return items;

  itemsCopy.splice(index, 1);
  return [...itemsCopy, items[index]];
};

export const formatPhoneNumberForPrintLetter = (phone: string) => {
  const cleaned = ('' + phone).replace(/\D/g, '');
  const match = cleaned.match(/^(\d{3})(\d{3})(\d{4})$/);

  if (match) {
    return '(' + match[1] + ') ' + match[2] + '-' + match[3];
  }

  return null;
};

export const getLatestItem = (items?: any): any => {
  if (!Array.isArray(items)) return undefined;
  if (items.length > 0) {
    return items[items.length - 1];
  }

  return undefined;
};

export const formatPhoneNumber = (phone: string, isMasked?: boolean) => {
  if (phone && isMasked) return `(***) ***-****`;
  if (!phone || phone.includes('-')) return phone;

  return `(${phone.slice(0, 3)}) ${phone.slice(3, 6)}-${phone.slice(6)}`;
};

export const getSelectedStore = () =>
  sessionStorage.getItem(STORE_NUMBER_KEY) || '';

export const randomNumericStringGenerator = (numberOfDigits = 8) => {
  if (typeof numberOfDigits !== 'number') {
    throw new TypeError(
      `Expected number instead got: ${typeof numberOfDigits}`
    );
  }

  return crypto
    .randomBytes(4)
    .readUInt32BE(0)
    .toString()
    .substr(0, numberOfDigits);
};

export const b64toBlob = (
  b64Data: any,
  contentType = 'application/pdf',
  sliceSize = 512
) => {
  const byteCharacters = atob(b64Data);
  const byteArrays = [];

  for (let offset = 0; offset < byteCharacters.length; offset += sliceSize) {
    const slice = byteCharacters.slice(offset, offset + sliceSize);

    const byteNumbers = new Array(slice.length);
    for (let i = 0; i < slice.length; i++) {
      byteNumbers[i] = slice.charCodeAt(i);
    }

    const byteArray = new Uint8Array(byteNumbers);
    byteArrays.push(byteArray);
  }

  return new Blob(byteArrays, { type: contentType });
};

export const formatMoney = (amount: string | number) => {
  if (amount === null || amount === undefined || isNaN(amount as number)) {
    return '$0.00';
  }

  return `$${Number(amount).toFixed(2)}`;
};

export const getContentType = (fileName: string, fileContentType?: string) => {
  if (fileContentType) return fileContentType;
  let result;

  if (fileName && typeof fileName === 'string') {
    const splits = fileName.split('.');
    const extension = splits[splits.length - 1] as string;
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    //@ts-ignore
    result = CONTENT_TYPE[extension];
  }

  return result || CONTENT_TYPE.pdf;
};

export const SIX_MONTHS_AGO = sub(new Date(), { months: 6 });
export const SIX_MONTHS_AFTER = add(new Date(), { months: 6 });

export const thirtyDaysBeforeToDate = (toDate: Date) =>
  sub(toDate, { days: 31 });
export const thirtyDaysAfterFromDate = (fromDate: Date) =>
  add(fromDate, { days: 31 });

export const spanishMonths: any = {
  '01': 'enero',
  '02': 'febrero',
  '03': 'marzo',
  '04': 'abril',
  '05': 'mayo',
  '06': 'junio',
  '07': 'julio',
  '08': 'agosto',
  '09': 'septiembre',
  '10': 'octubre',
  '11': 'noviembre',
  '12': 'diciembre',
};
