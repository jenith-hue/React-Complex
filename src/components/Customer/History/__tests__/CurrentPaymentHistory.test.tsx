/* eslint-disable sonarjs/no-duplicate-string */
/* eslint-disable react/display-name */
/* eslint-disable sonarjs/no-identical-functions */

import React from 'react';
import renderer from 'react-test-renderer';
import { CurrentPaymentHistory } from '../CurrentPaymentHistory';
import * as Utils from '../../../../utils/utils';

interface PropWithChildren {
  children: JSX.Element;
}

interface MockRACTableProps {
  renderTableHead: () => JSX.Element;
  renderTableContent: () => JSX.Element;
}

jest.mock(
  'react-infinite-scroll-component',
  () =>
    ({ children, ...rest }: PropWithChildren) =>
      (
        <div data-testid="InfiniteScroll" {...rest}>
          {children}
        </div>
      )
);

jest.mock('@rentacenter/racstrap', () => ({
  ...jest.requireActual('@rentacenter/racstrap'),
  RACButton: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="RACButton" {...rest}>
      {children}
    </div>
  ),
  Typography: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="Typography" {...rest}>
      {children}
    </div>
  ),
  RACTableCell: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="RACTableCell" {...rest}>
      {children}
    </div>
  ),
  RACTableRow: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="RACTableRow" {...rest}>
      {children}
    </div>
  ),
  RACCOLOR: {
    INDEPENDENCE: 'INDEPENDENCE',
  },
  RACTable: ({ renderTableHead, renderTableContent }: MockRACTableProps) => (
    <div data-testid="RACTable">
      <div id="table-head">{renderTableHead()}</div>
      <div id="table-content">{renderTableContent()}</div>
    </div>
  ),
  makeStyles: () => () => ({
    tableContentColor: 'tableContentColor',
    tableRoot: 'tableRoot',
    tableBody: 'tableBody',
    tableHead: 'tableHead',
    contentHeight: 'contentHeight',
  }),
}));

jest.mock('../../../../context/PaymentHistory/PaymentHistoryProvider', () => ({
  usePaymentHistoryActions: () => ({
    fetchPaymentHistory: jest.fn(),
    resetPaymentData: jest.fn(),
  }),
  usePaymentHistory: () => ({
    total: 20,
    numberOfLoadedPayments: 20,
    hasApiError: false,
    loading: false,
    payments: [
      {
        storeNumber: jest.fn(),
        paymentDate: '2021-05-08',
        receiptDate: '2021-05-08T12:54:27',
        receiptId: '9999646083762',
        receiptTransactionType: {
          code: 'P',
          description: 'Payment',
        },
        paymentAmount: '116.98',
        nextDueDate: '2021-06-08',
        agreementId: '1234567',
        agreementNumber: '08976767956',
        daysExtension: '0',
        daysLate: '0',
        paymentOrigin: {
          code: 'A',
          description: 'AutoPay',
        },
        paymentMethodType: ['CC', 'CASH'],
        netRentalRevenue: '92.77',
        clubPayment: '6.00',
      },
    ],
  }),
}));

// Notice that the below is required, even if there is an another
// jest.spyOn in the test.
// Otherwise, we'll get the following: TypeError: Cannot redefine property: useLocation
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: '',
  }),
}));
describe('CurrentPaymentHistory', () => {
  it('renders correctly', () => {
    jest
      .spyOn(Utils, 'formatDateString')
      .mockImplementation(() => '12/12/2021');
    jest
      .spyOn(Utils, 'formatStringDateHoursAndMinutes')
      .mockImplementation(() => '10:00 AM');

    const tree = renderer.create(<CurrentPaymentHistory />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
