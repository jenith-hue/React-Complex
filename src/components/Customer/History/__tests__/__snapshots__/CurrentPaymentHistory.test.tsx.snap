// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentPaymentHistory renders correctly 1`] = `
Array [
  "",
  <div
    data-testid="InfiniteScroll"
    dataLength={20}
    hasMore={false}
    height={330}
    loader={<Loader />}
    next={[MockFunction]}
  >
    <div
      data-testid="RACTable"
    >
      <div
        id="table-head"
      >
        <div
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Date paid
          </div>
        </div>
        <div
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Day
          </div>
        </div>
        <div
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Time
          </div>
        </div>
        <div
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Receipt #
          </div>
        </div>
        <div
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Total Payment
          </div>
        </div>
        <div
          align="center"
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Next Due Date
          </div>
        </div>
        <div
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Days Ext
          </div>
        </div>
        <div
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            # Days late
          </div>
        </div>
        <div
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Store
          </div>
        </div>
        <div
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Payment Origin
          </div>
        </div>
        <div
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Tender Type
          </div>
        </div>
        <div
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Net Rent
          </div>
        </div>
        <div
          data-testid="RACTableCell"
        >
          <div
            data-testid="Typography"
            variant="h5"
          >
            Club
          </div>
        </div>
      </div>
      <div
        id="table-content"
      >
        <div
          backgroundColor="white"
          data-testid="RACTableRow"
          hover={true}
        >
          <div
            data-testid="RACTableCell"
          >
            <div
              className="tableContentColor"
              data-testid="Typography"
              variant="body2"
            >
              12/12/2021
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              className="tableContentColor"
              data-testid="Typography"
              variant="body2"
            >
              Sat
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              className="tableContentColor"
              data-testid="Typography"
              variant="body2"
            >
              10:00 AM
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              color="primary"
              data-testid="RACButton"
              onClick={[Function]}
              size="small"
              variant="text"
            >
              <div
                color="primary"
                data-testid="Typography"
                onClick={[Function]}
                variant="body2"
              >
                9999646083762
              </div>
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              className="tableContentColor"
              data-testid="Typography"
              variant="body2"
            >
              $116.98
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              align="center"
              className="tableContentColor"
              data-testid="Typography"
              variant="body2"
            >
              12/12/2021
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              className="tableContentColor"
              data-testid="Typography"
              variant="body2"
            >
              0
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              className="tableContentColor"
              data-testid="Typography"
              variant="body2"
            >
              0
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              className="tableContentColor"
              data-testid="Typography"
              variant="body2"
            />
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              className="tableContentColor"
              data-testid="Typography"
              variant="body2"
            >
              AutoPay
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              className="tableContentColor"
              data-testid="Typography"
              variant="body2"
            >
              CC, CASH
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              className="tableContentColor"
              data-testid="Typography"
              variant="body2"
            >
              $92.77
            </div>
          </div>
          <div
            data-testid="RACTableCell"
          >
            <div
              className="tableContentColor"
              data-testid="Typography"
              variant="body2"
            >
              $6.00
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
